statsmodels/tsa/stattools.py

```diff
@@ -1896,6 +1896,100 @@ def arma_order_select_ic(
     return Bunch(**res)
 
 
+def diebold_mariano_test(y, forecast1, forecast2, horizon=1,
+                         criterion="MSE", power=2, harvey_adj=False):
+    """
+    Performs a Diebold-Mariano test under the null hypothesis of
+    equal predictive accuracy between two forecasts.
+
+    Parameters
+    ----------
+    y : array_like
+        Array of the observed variable.
+    forecast1 : array_like
+        Array of the forecasted values.
+    forecast2 : array_like
+        Array of the forecasted values.
+    horizon : int
```

Comment:
Would be really good to have an example showing how horizon=2 is really used.  What alignment is assumed for Y and the forecasts? For example, if horizon=5, y[100] is observation 100 for y.  What is f1[100] and f1[105]?